Deleting all Set-Cookie headers in the response

Doerge opened the pull request:
All Set-Cookie headers are now deleted on the response, before the zappa cookie is set.

Added comments to make it (hopefully) make it clearer what is going on.

bbangert commented:
LGTM

Miserlou commented:
Looking good, but do I need to merge before we have the expiry working? 

Also - since this is critical and complicated code, a test would be GREATLY appreciated..

Doerge commented:
There were some  ugly stuff going on in the middleware. Rewrote most of the logic and added 700 lines of tests.

I still have trouble reproducing bugs from the production environment, though :8ball:

collingreen commented on tests/tests_middleware.py:
```diff
@@ -0,0 +1,356 @@
+        self.assertDictEqual(received, excpected)
+
+    def test_wsgi_middleware_cookieoverwrite(self):
+        """ This method do:
```
nitpick:  remove 'do'

collingreen commented on tests/tests_middleware.py:
```diff
@@ -0,0 +1,356 @@
+        self.assertTrue(zappa_cookie.startswith('zappa='))
+
+        # Reads the hopefully decoded cookies
+        def simple_app(environ, start_response):
```
nitpick - redefined simple_app in this test
